### PR TITLE
feat: trustworthy schema/flow editing journey (guidance, clean diffs, single-apply re-pin)

### PR DESCRIPTION
**Pull request by fforootd**
## Summary

Fixes the UX failures from the early schema-editing testing round (Elina's journey): the post-setup dead-end, scaffold↔live drift with silent-stripping risk, noisy plan diffs, and the `E_VALIDATION: flow field: not a property in the user schema` blocker when a schema edit and a flow referencing the new field land in the same `apply`.

Four commits, one per track (splittable):

- **Guidance & docs** (`2b36c1c`): the "Zitadel is ready" next steps now surface the editable `.zitadel/schemas|flows` files, their READMEs, and `plan`/`apply` (human + `--json`). Scaffolded READMEs rewritten workflow-first (mental model → example → making changes → common changes, per Elina's draft). Corrects the stale `concepts/flows-and-schemas.mdx` claim that setup doesn't scaffold; adds the add-`company`-field walkthrough to `cli/plan-apply.mdx`; quickstart gains a "Customize the login" section.
- **Diff parity & noise** (`a258697`): new `@zitadel/config/normalize` strips server-echoed noise (empty `audience`) and spelled-out `x-*` meta-schema property defaults. State hashing becomes normalized + key-order-insensitive (with a legacy-hash fallback so existing `state.json` files never read as edits), diffs render both sides normalized, and setup/apply write the server's canonical body back to the local file (churn-guarded, reported in output and `files_updated`). Scaffolded files now match live state by construction; the api-mock mirrors the server's unconditional `audience: {}` echo so the parity integration test is honest.
- **Auto-repin in one apply** (`a993e55`): a schema revise records the superseded id (`previousId` in `state.json`), rewrites `user_schema` in every pinned local flow file (format-preserving), and the same-run flow update carries the new revision id — schema + flow changes land in a single `apply`. Untouched-but-pinned flows get an explicit repin update; `plan` announces the re-pin instead of printing manual instructions; interrupted runs recover via `previousId`, which is cleared once nothing pins the old revision so deliberately pinning an old revision is never force-bumped.
- **Renderer fixes found in live verification** (`69bb95a`): array diffs compared order-sensitively (server field order vs locally sorted files rendered unchanged `steps` as full churn); plan's "N unchanged" was arithmetically always 0.

Design decision (after codex review of the plan): local config stays authoritative — per-file uploads, with server-provided ids propagated back into state and referencing files (git-visible, plan-announced). A server-side atomic bundle-apply API was designed and deliberately deferred as the first slice of #449; #445's per-flow opt-in pinning is superseded for local-first projects until releases land.

## Validation

- `moon run cli:test` — 710/710 (includes new coverage: legacy-hash fallback, write-back churn guard, repin-only updates, crash recovery, URL-form ids, plan-writes-no-files, and an integration test replaying the full journey against the api-mock)
- `moon run config:test cli:typecheck cli:lint config:typecheck config:lint api-mock:build` — all green
- Live replay against a real Go server built from this branch (embedded postgres, `start --runtime binary`): fresh setup → `plan` immediately empty → add optional `company` to the schema → plan shows exactly one `+ company` plus the re-pin announcement, zero `audience`/`x-editable` noise → add `company` to the register step `fields` → **one `apply` succeeds** (no E_VALIDATION), flow file/state/live server all pin the new revision → second `plan` empty.

## Release notes / changeset

- Changeset: `.changeset/setup-customize-guidance.md` — setup guidance + README restructure (@zitadel/cli, @zitadel/config)
- Changeset: `.changeset/normalize-sync-parity.md` — normalizer, clean diffs, write-back (@zitadel/config, @zitadel/cli)
- Changeset: `.changeset/apply-auto-repin.md` — single-apply re-pin (@zitadel/cli, @zitadel/config)

## Notes

- Reviewer focus: the hash-migration fallback in `buildSyncPlan` (a spurious mismatch would publish a garbage revision — covered by tests) and the revise-branch ordering in `runSyncLoop` (state with `previousId` advances before flow rewrites, so a rerun never double-publishes).
- Normalized bodies are comparison/local-file only — upload payloads stay raw, because the server does not materialize meta-schema defaults (stripping `"x-editable": true` from the wire would silently drop it).
- Follow-ups spotted but out of scope (happy to file as issues): duplicate primary-passkey button in `default.liquid` (spun off separately), action name/kind three-layer coupling, flow presets, `presets.ts` `"x-mfa": "sms"` vs boolean meta-schema default, `SchemaSyncer.delete` throwing mid-apply, optional Go hardening to stop echoing empty `audience`.

### CODE REVIEW

**Comment by fforootd** [comment]
Addressed the codex review in 0797c4e:

**Finding 1 (write-back strips schema settings) — confirmed and fixed.** The compare-normalizer was reused for file write-back, so a canonical schema carrying `"x-editable": true` would land stripped in the local file and the next revise would publish without it. Write-back now has a separate `normalizeWrite` hook: flows keep stripping pure transport noise (detail-envelope keys, the empty `audience` echo — semantically absent either way), while schema canonical bodies are written **verbatim**; `normalize` is comparison-only (hash/diff). Covered by new tests (`write-back preserves spelled-out schema defaults`, updated setup-resources write-back test).

**Finding 2 (auto-repin of untouched flows vs #445) — behavior kept by explicit owner decision, failure mode closed.** The local-authoritative model (repo files are truth; server-assigned ids propagate back into state and referencing files, plan-announced and git-visible) supersedes #445's per-flow opt-in for local-first projects until #449 releases land — that trade-off is documented in the PR description. The concrete hazard you raised (property rename/removal → untouched pinned flow fails server validation *mid-apply*, after the revision published) is real and now impossible: `buildSyncPlan` pre-flight-validates every flow adopting a new revision — each plain `steps[].fields` entry must exist in the edited schema's `properties` (credential tokens like `x-auth-methods#…` excluded) — so both `plan` and `apply` fail fast with the flow and field named, before any platform mutation. Nothing half-applies.

**Finding 3 (changesets check) — fixed.** `@zitadel/api-mock` removed from `.changeset/normalize-sync-parity.md` (it's on the `ignore` list, and ignored packages don't take changesets). Also retitled the PR without the `cli` scope per `.github/semantic.yml`.

Suites after the changes: 713/713 CLI tests, config 13/13, typecheck + lint clean.

File: apps/cli/src/lib/sync/loop.ts [hunk 1]
```diff
@@ -135,39 +188,85 @@ export async function buildSyncPlan(
...
       case "create": {
-        const id = await action.syncer.create(action.content);
-        const entry: ResourceEntry = { id, hash: action.hash };
+        const { id, canonical } = await action.syncer.create(action.content);
+        const entry: ResourceEntry = { id, hash: await writeBack(action, canonical, action.hash) };
         await updateState(cwd, action.path, entry);
         consola.info(
           `Created a new ${action.syncer.kind} on Zitadel from ${action.path} (id ${id})`,
         );
         break;
```
[the remark this answers is not in the document]
Comment (fforootd) [reply apps/cli/src/lib/sync/loop.ts:263]: Confirmed and fixed in d1571675 — the create branch returned before the repin computation, so a new flow POSTed the stale pin and its canonical echo could revert the file the revise had just re-pinned. `buildSyncPlan` now computes/attaches `repin` before the create branch (pre-flight field validation included), and the executor patches `user_schema` before POSTing, with the fallback hash computed on the patched content. Covered by two new tests: same-run revise + new flow POSTs the minted id and converges, and a stale-echo write-back cannot resurrect the old pin.

File: apps/cli/src/lib/sync/loop.ts [hunk 2]
```diff
@@ -72,20 +88,43 @@ export async function buildSyncPlan(
...
+      // State files written before normalized hashing hold legacy hashes
+      // (order-sensitive, un-normalized). Accepting either format keeps an
+      // untouched file a skip — a spurious mismatch here would publish a
+      // garbage schema revision. Writes always store the new format, so
+      // state converges on the next real change.
+      const unchanged = entry.hash === hash || entry.hash === hashResourceContent(content);
+      if (unchanged && !(repin && syncer.mutable)) {
         actions.push({ kind: "skip", path: relPath, reason: "no-change" });
         continue;
       }
```
[the remark this answers is not in the document]
Comment (fforootd) [reply apps/cli/src/lib/sync/loop.ts:151]: Fixed in d1571675 — the skip gate now also accepts the legacy hash of the stably-sorted content (`hashResourceContent(JSON.parse(stableStringify(content)))`), which is exactly how setup-era state hashes were computed, so upgrade-then-reorder no longer schedules a spurious update/revision. The remaining uncovered case is a legacy hash that was itself computed over hand-ordered keys *and* the user reorders again before the first new-format apply — that first apply is content-idempotent and converges state to the order-insensitive format, so the window closes itself. Regression test added.

File: apps/cli/src/lib/sync/loop.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fforootd) [reply apps/cli/src/lib/sync/loop.ts:263]: Fixed in d1571675: buildSyncPlan now attaches the repin to flow creates too (with pre-flight field validation), and the executor patches user_schema before POSTing — so a new flow can't carry the stale pin and its canonical echo can't revert the re-pinned file. Two regression tests added.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: apps/cli/src/lib/sync/loop.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fforootd) [reply apps/cli/src/lib/sync/loop.ts:151]: Fixed in d1571675: the skip gate additionally accepts the legacy hash of the stably-sorted content (how setup-era hashes were computed), so upgrade-then-reorder no longer schedules a spurious revision; regression test added.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by vitorbari** [comment]
I tested the changes and it works great. There is an edge-case, if I update the local schema with a change that is equals the default value stored in the server, then the CLI shows no diff. e.g. adding `"x-editable": true,` to a field. I guess it is fine because it is not changing any behaviour, but in general, it makes me wonder if we should continue storing the normalized schemas, or should store the actual raw schema the user provides.

**Comment by fforootd** [comment]
Good catch — and worth spelling out, because this changed mid-review: since 0797c4e we **do** store the raw schema the user provides, everywhere. Write-back keeps schema bodies verbatim (no normalization), and upload payloads are never normalized either — so your `"x-editable": true` stays byte-for-byte in the local file and is included as written the next time a real edit publishes a revision. Nothing normalized is ever persisted.

Normalization only exists in the **comparison** layer (state hash + plan diff). That's exactly why your edit shows no diff: per the meta-schema, `x-editable` defaults to `true`, so spelling it out changes no behavior — and treating it as a change would be the mirror image of the noise Elina hit originally (a live schema spelling out `x-editable: true` against a local file omitting it rendered as a scary red removal). Same semantic equivalence, honored in both directions.

Added a note to the plan-and-apply docs page making this explicit: files are stored as written, comparison is semantic, and default-only edits ride along on the next real publish.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by vitorbari** [comment]
> Good catch — and worth spelling out, because this changed mid-review: since [0797c4e](https://github.com/zitadel/nextgen/commit/0797c4eaaed80d7b4a53961a84539f088f110b9c) we **do** store the raw schema the user provides, everywhere. Write-back keeps schema bodies verbatim (no normalization), and upload payloads are never normalized either — so your `"x-editable": true` stays byte-for-byte in the local file and is included as written the next time a real edit publishes a revision. Nothing normalized is ever persisted.
> 
> Normalization only exists in the **comparison** layer (state hash + plan diff). That's exactly why your edit shows no diff: per the meta-schema, `x-editable` defaults to `true`, so spelling it out changes no behavior — and treating it as a change would be the mirror image of the noise Elina hit originally (a live schema spelling out `x-editable: true` against a local file omitting it rendered as a scary red removal). Same semantic equivalence, honored in both directions.
> 
> Added a note to the plan-and-apply docs page making this explicit: files are stored as written, comparison is semantic, and default-only edits ride along on the next real publish.
> 
> _🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

user-schemas are stored normalized under the `json_schemas` table. The part I'm missing, is if this is intentional behavior in the `POST /schemas` endpoint.

**Comment by fforootd** [comment]
> user-schemas are stored normalized under the `json_schemas` table. The part I'm missing, is if this is intentional behavior in the `POST /schemas` endpoint.

Yes, intentional-in-effect: POST stores the ogen-decoded document with meta-schema defaults materialized, which keeps immutable revisions self-contained; local files stay as the author wrote them and the CLI compares semantically across the two forms.

**Review by vitorbari — APPROVED** [review]
